pkgdb/src/buildenv/realise.cc

```diff
@@ -80,13 +80,9 @@ then
 fi
 
 if [ -d "$FLOX_ENV/etc/profile.d" ]; then
-  declare -a _prof_scripts;
-  _prof_scripts=( $(
-    shopt -s nullglob;
-    echo "$FLOX_ENV/etc/profile.d"/*.sh;
-  ) );
-  for p in "${_prof_scripts[@]}"; do . "$p"; done
-  unset _prof_scripts;
+  while IFS= read -r -d '' profile_script; do
+    . "$profile_script"
+  done < <(find "$FLOX_ENV/etc/profile.d" -name '*.sh' -print0)
```

Comment:
This introduces a dependency on `find` and correspondingly the `findutils` package which is not currently included in the closure. So we either need to fix that or we can take the simpler approach of doing something like this:
```
if [ -d "$FLOX_ENV/etc/profile.d" ]; then
  declare -a _prof_scripts;
  _prof_scripts=( $(
    cd "$FLOX_ENV/etc/profile.d";
    shopt -s nullglob;
    echo *.sh;
  ) );
  for p in "${_prof_scripts[@]}"; do . "$FLOX_ENV/etc/profile.d/$p"; done
  unset _prof_scripts;
fi
```
_(FWIW I've tested that this works.)_

Thanks.